homeassistant/components/bluetooth/__init__.py

```diff
@@ -213,6 +222,32 @@ async def async_get_adapter_from_address(
     return await _get_manager(hass).async_get_adapter_from_address(address)
 
 
+async def _async_check_haos_version(hass: HomeAssistant) -> None:
+    """Check if the version of Home Assistant Operating System is new enough."""
+    if not any(
+        entry
+        for entry in hass.config_entries.async_entries(DOMAIN)
+        if entry.source != SOURCE_IGNORE
+    ):
+        return
+
+    system_info = await async_get_system_info(hass)
+    if (
+        not (haos_version := system_info.get("hassos"))
```

Comment:
Need to make sure this is the right key

Reply:
Something is wrong here because I'm getting this on my HAOS box

```
 [homeassistant.components.bluetooth] System info: {'installation_type': 'Home Assistant Container', 'version': '2022.10.0.dev0', 'dev': True, 'hassio': False, 'virtualenv': False, 'python_version': '3.10.5', 'docker': True, 'arch': 'aarch64', 'timezone': 'America/Chicago', 'os_name': 'Linux', 'os_version': '5.15.67', 'user': 'root'}
```

Reply:
bluetooth still gets setup before hassio even if I add it after_deps

Reply:
```
2022-09-16 06:42:06.192 WARNING (MainThread) [homeassistant.components.bluetooth] System info: {'installation_type': 'Home Assistant OS', 'version': '2022.10.0.dev0', 'dev': True, 'hassio': Tr
ue, 'virtualenv': False, 'python_version': '3.10.5', 'docker': True, 'arch': 'aarch64', 'timezone': 'America/Chicago', 'os_name': 'Linux', 'os_version': '5.15.67', 'user': 'root', 'supervisor'
: '2022.09.dev1602', 'host_os': 'Home Assistant OS 10.0.dev20220912', 'docker_version': '20.10.17', 'chassis': 'embedded'}
```

Might need to use a different api